sni.go

```diff
@@ -17,21 +17,40 @@ package tcpproxy
 import (
 	"bufio"
 	"bytes"
+	"context"
 	"crypto/tls"
 	"io"
 	"net"
+	"strings"
 )
 
 // AddSNIRoute appends a route to the ipPort listener that says if the
 // incoming TLS SNI server name is sni, the connection is given to
 // dest. If it doesn't match, rule processing continues for any
 // additional routes on ipPort.
 //
+// By default, the proxy will route all ACME tls-sni-01 challenges
+// received on ipPort to all SNI dests. You can disable ACME routing
+// with StopAcme.
```

Comment:
StopACME per https://github.com/golang/go/wiki/CodeReviewComments#initialisms

(that's why it's not AddSniRoute)